File: scripts/check_db.py
import socket
import sys

host = '127.0.0.1'
port = 30314

try:
    s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
    s.settimeout(5)
    s.connect((host, port))
    print(f"Connected to {host}:{port}")
    
    # MySQL server sends a handshake packet first
    data = s.recv(1024)
    print(f"Received {len(data)} bytes")
    print(f"Hex: {data.hex()[:100]}...")
    print(f"Raw: {data[:100]}")
    
    s.close()
except Exception as e:
    print(f"Error: {e}")

File: scripts/check_db.js
const net = require('net');

const client = new net.Socket();
const port = 30314;
const host = '127.0.0.1';

client.connect(port, host, function() {
    console.log('Connected');
});

client.on('data', function(data) {
    console.log('Received: ' + data.length + ' bytes');
    console.log('Hex: ' + data.toString('hex').substring(0, 100));
    console.log('String: ' + data.toString().substring(0, 100));
    client.destroy();
});

client.on('close', function() {
    console.log('Connection closed');
});

client.on('error', function(err) {
    console.error('Error: ' + err.message);
});
